### PR TITLE
new recipe for melon/1.0.0-alpha.1

### PR DESCRIPTION
**Pull request by fhamonic**
### Summary
Changes to recipe:  **melon/1.0.0-alpha.1**

#### Motivation
Add melon, a C++20 high performance graph library.

#### Details
https://github.com/fhamonic/melon

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/melon/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            # Validate the minimum cpp standard supported when installing the package. For C++ projects only
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        # In case this library does not work in some another configuration, it should be validated here too
+        if self.settings.os == "Windows":
```
Comment (AbrilRBS) [inline recipes/melon/all/conanfile.py:74]: Is this a real condition? or did you forget to remove the example from the package template?

File: recipes/melon/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    def layout(self):
+        # src_folder must use the same source folder name than the project
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        # Prefer self.requires method instead of requires attribute
+        # Direct dependencies of header only libs are always transitive since they are included in public headers
+        self.requires("range-v3/[>=0.11.0]", transitive_headers=True)
+        self.requires("fmt/[>=10.0.0]", transitive_headers=True)
```
Comment (AbrilRBS) [inline recipes/melon/all/conanfile.py:57]: `transitive_headers` not needed for `header-only` libraries, it's assumed True by default for these kind of packages

File: recipes/melon/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        # self.cpp_info.set_property("cmake_module_file_name", "melon")
+        # self.cpp_info.set_property("cmake_module_target_name", "melon::melon")
+
+        # Set these to the appropriate values if package provides a CMake config file
+        # (package-config.cmake or packageConfig.cmake, with package::package target, usually installed in <prefix>/lib/cmake/<package>/)
+        # self.cpp_info.set_property("cmake_file_name", "melon")
+        # self.cpp_info.set_property("cmake_target_name", "melon::melon")
+        # Set this to the appropriate value if the package provides a pkgconfig file
+        # (package.pc, usually installed in <prefix>/lib/pkgconfig/)
+        # self.cpp_info.set_property("pkg_config_name", "melon")
```
Comment (AbrilRBS) [inline recipes/melon/all/conanfile.py:113]: Feel free to remove these lines, alongside the rest of the comments from the template :)

File: recipes/melon/all/conandata.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "1.0.0-alpha.1":
```
Comment (AbrilRBS) [inline recipes/melon/all/conandata.yml:2]: Do we have an idea on when proper `1.0.0` would be released?

**Review by AbrilRBS — COMMENTED** [review]
Hi @fhamonic thanks a lot for taking the time to create this PR, we appreciate it! I have a few minor comments, but overall this looks great! :)

**Comment by fhamonic** [comment]
Hi @AbrilRBS, thanks for taking the time to review this recipe and thanks for your comments!
I've done the modifications and pushed :)

File: recipes/melon/all/conanfile.py (hunk 1 above)
Comment (fhamonic) [reply recipes/melon/all/conanfile.py]: You are right, I forgot..

File: recipes/melon/all/conandata.yml (hunk 4 above)
Comment (fhamonic) [reply recipes/melon/all/conandata.yml:2]: Actually, algorithms are well tested but I feel like the library lacks some features like serialization/visualization to be complete. I have been asked to submit a recipe and I think that making the library available on Conan-center could stimulate potential feedback and maybe contributions for designing these features.
Given that this is an important project for me, I'll continue working on it for sure :)

Comment (AbrilRBS) [reply recipes/melon/all/conandata.yml:2]: Great to hear! Thanks for the extra insight

**Review by AbrilRBS — APPROVED** [review]
I've pinned the dependencies versions, as those are not quite ready to be used as version ranges yet, sorry for the confussion!

Otherwise, this looks great now, thanks a lot for taking the time to get the PR thru to the finish line, we appreciate it :)

**Review by perseoGI — APPROVED** [review]
LGTM! Thanks a lot for contributing to conan! And the library looks promising!
It would be great to add some conan instructions to your repo README!
Something similar to: https://github.com/jlblancoc/nanoflann#using-conan 🚀

**Pull request merged**